src/templateHandler.ts

```diff
@@ -68,6 +68,9 @@ export class TemplateHandler {
         };
         await this.compiler.compile(document, scopeData, context);
 
+        // execute each extension in the supplied order
+        await this.callExtensions<T>(scopeData, context);
```

Comment:
I wonder, as you commented before maybe it would be better if it was possible to execute the extensions in different stages of the "process"? I think I'd prefer something like `options.extensions.beforeCompilation` and `options.extensions.afterCompilation`.

Reply:
Will implement